lib/types/json.js

```diff
@@ -149,12 +150,12 @@ function json (options) {
  * @private
  */
 
-function createStrictSyntaxError (str, char) {
+function createStrictSyntaxError (parser, str, char) {
   var index = str.indexOf(char)
   var partial = str.substring(0, index) + '#'
 
   try {
-    JSON.parse(partial); /* istanbul ignore next */ throw new SyntaxError('strict violation')
+    parser(partial); /* istanbul ignore next */ throw new SyntaxError('strict violation')
```

Comment:
This is the line my above comment was supposed to be on.

Reply:
I'm not sure this is right for custom parsers. Or at least it's not clear in the docs for the parser option that this is going to happen and how people should build their parser function to compensate.